src/main/java/org/reactome/qa/doisuggester/Main.java

```diff
@@ -50,213 +56,201 @@ public static void main(String[] args) throws FileNotFoundException, IOException
-											// If 'Pathway' key exists, add to it
-											if (pathway2Reactions2NewIEs.get(parentPathwayWithRLEInstanceEdit) != null)
-											{
-												pathway2Reactions2NewIEs.get(parentPathwayWithRLEInstanceEdit).add(reaction2newIE);
-											}
-											else
-											{
-												// If 'Pathway' key doesn't exist, create it and then add to it.
-												Set<Map<GKInstance, GKInstance>> reaction2newIESet = new HashSet<>(Arrays.asList(reaction2newIE));
-												pathway2Reactions2NewIEs.put(parentPathwayWithRLEInstanceEdit, reaction2newIESet);
-											}
-										}
-										else
-										{
-											// If no Pathways in the hierarchy have the new IE, then the program defaults to
-											// outputting the immediate parent Pathway of the RlE
-											Map<GKInstance, GKInstance> reaction2newIE = new HashMap<>();
-											reaction2newIE.put(currentRLE, newIE);
-											// If 'Pathway' key exists, add to it
-											if (pathway2Reactions2NewIEs.get(parentPathway) != null)
-											{
-												pathway2Reactions2NewIEs.get(parentPathway).add(reaction2newIE);
-											}
-											else
-											{
-												// If 'Pathway' key doesn't exist, create it and then add to it.
-												Set<Map<GKInstance, GKInstance>> reaction2newIESet = new HashSet<>(Arrays.asList(reaction2newIE));
-												pathway2Reactions2NewIEs.put(parentPathway, reaction2newIESet);
-											}
-										}
-									}
-								}
-							}
-						}
+						pathway2Reactions2NewIEs = processPreexistingRLEs(pathway2Reactions2NewIEs, currentRLE, previousRLE);
 					}
+					
 				}
 			}
 
-			// For cases where this the first Release that a RlE has appeared (ie. fetching
-			// for the RlE in the previous DB returned null).
+			Map<GKInstance, Set<Map<GKInstance, GKInstance>>> pathwaysMap = processNewRLEs(newRLEs, pathway2Reactions2NewIEs);
 
-			// As before, iterate through all RlEs.
-			for (GKInstance newRLE : newRLEs)
-			{
-				// Get non-Reactome IEs from the revised, authored, and reviewed attributes.
-				List<GKInstance> newRLEInstanceEdits = getNonReactomeEventIEs(newRLE);
+			printOutput(pathwaysMap);
 
-				// Look through all parents of the RlE, trying to find the new instance edit in
-				// any of their revised, authored, or reviewed attributes.
-				List<GKInstance> parentPathways = (List<GKInstance>) newRLE.getReferers(ReactomeJavaConstants.hasEvent);
-				for (GKInstance parentPathway : parentPathways)
-				{
-					// Get all non-Reactome instance edits in the revised, authored, and reviewed
-					// attributes of the Pathway.
-					List<GKInstance> parentPathwayIEs = getNonReactomeEventIEs(parentPathway);
+		}
+		catch (SQLException e)
+		{
+			e.printStackTrace();
+		}
+		catch (InvalidAttributeException e)
+		{
+			// TODO Auto-generated catch block
+			e.printStackTrace();
+		}
+		catch (Exception e)
+		{
+			// TODO Auto-generated catch block
+			e.printStackTrace();
+		}
```

Comment:
These can be joined as one:
```
catch (SQLException | InvalidAttributeException | Exception e) {
    e.printStackTrace();
}
```

Reply:
Yes, I know that. That will probably happen later, unless I decide to have more specific exception-handling.

Reply:
Why not do it now with the other clean-up?  All 3 blocks do the same thing, so they can be combined.

Reply:
The original cleanup of this PR  is really about taking a few very large, oversized functions and breaking them up into smaller pieces so that they are easier to follow. There was also a lot of repetitive code that was refactored into methods.

The code was already like this and I don't see any problem in leaving it this way - it will be cleaned up in future PRs but for now it doesn't cause a problem so I'm leaving it.